docs/agents/distributed-tracing.md

```diff
@@ -0,0 +1,63 @@
+# Distributed Tracing
+
+We implement the W3C standards, both for HTTP headers and binary fields.
+
+To test compatibility with the standard,
+agents MUST validate against the w3c [test harness](https://github.com/w3c/trace-context/blob/master/test/test_data.json).
+
+## Http Headers
+
+Our implementation relies on the [W3C Trace Context](https://www.w3.org/TR/trace-context-1/) standard. Until this standard became final,
+we used the header name `elastic-apm-traceparent` and we did not support `tracestate`. Soon after the standard became official, we
+started to fully align with it. For backward compatibility reasons, this was done in phases, so that the first step was to look for both
+`traceparent` headers in incoming requests (meaning - both `elastic-apm-traceparent` and `traceparent`) and sending both `traceparent` headers in outgoing requests (with the exception of the RUM agent,
+due to CORS). [Issue #71](https://github.com/elastic/apm/issues/71) describes this in more detail, as well as tracking implementation
+in the different agents.
+New agents may decide whether to support both `traceparent` headers (so to be compatible with older agent versions) or only the formal W3C
+header.
+
+## Tracestate
+
+For our own `elastic` `tracestate` entry we will introduce a `key:value` formatted list of attributes.
+This is used to propagate the sample weight downstream, for example.
+See the [sampling](sampling.md) specification for more details.
+
+The general `tracestate` format is:
+
+    tracestate: elastic=key:value;key:value...,othervendor=<opaque>
+
+For example:
+
+    tracestate: elastic=w:5,othervendor=<opaque>
+
+
+### Validation and length limits
+
+The [`tracestate`](https://www.w3.org/TR/trace-context/#tracestate-header)
+specification lists a number of validation rules.
+In addition to that,
+there are specific rules for the attributes under the `elastic` entry.
+
+Agents MUST implement these validation rules:
+
+- The `tracestate` field may contain a maximum of 32 entries.
```

Comment:
What do we do if there's already 32 entries, not including an `elastic` one?

Reply:
The spec says this: 

> Entries larger than 128 characters long SHOULD be removed first. Then entries SHOULD be removed starting from the end of tracestate.

Reply:
but not including `elastic` would be the simplest strategy